soc/nxp/imx/Kconfig.soc

```diff
@@ -0,0 +1,10 @@
+# Copyright 2017,2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_IMX
+	bool
+
+config SOC_FAMILY
+	default "nxp_imx" if SOC_FAMILY_IMX
```

Comment:
```suggestion
config SOC_FAMILY_NXP_IMX
	bool

config SOC_FAMILY
	default "nxp_imx" if SOC_FAMILY_NXP_IMX
```

Reply:
Fixed, thanks.

Reply:
I synced all to still use :SOC_FAMILY_SOC as it has been used for many SoC and many device drivers.

drivers/clock_control/clock_control_mcux_syscon.c:#elif defined(CONFIG_SOC_FAMILY_IMX)
modules/Kconfig.imx:    depends on SOC_FAMILY_IMX
modules/Kconfig.mcux:   depends on SOC_FAMILY_KINETIS || SOC_FAMILY_IMX || SOC_FAMILY_LPC || \
samples/drivers/ipm/ipm_imx/sample.yaml:    filter: CONFIG_SOC_FAMILY_IMX
soc/nxp/imx/Kconfig:config SOC_FAMILY_IMX
soc/nxp/imx/Kconfig.defconfig:if SOC_FAMILY_IMX
soc/nxp/imx/Kconfig.defconfig:endif # SOC_FAMILY_IMX
soc/nxp/imx/Kconfig.soc:config SOC_FAMILY_IMX
soc/nxp/imx/Kconfig.soc:        default "imx" if SOC_FAMILY_IMX
soc/nxp/imx/imx8m/Kconfig.soc:  select SOC_FAMILY_IMX
soc/nxp/imx/imx9/Kconfig.soc:   select SOC_FAMILY_IMX
soc/soc_legacy/arm/nxp_imx/Kconfig:config SOC_FAMILY_IMX
soc/soc_legacy/arm/nxp_imx/Kconfig:if SOC_FAMILY_IMX
soc/soc_legacy/arm/nxp_imx/Kconfig:endif # SOC_FAMILY_IMX
soc/soc_legacy/arm/nxp_imx/mcimx6x_m4/Kconfig.series:   select SOC_FAMILY_IMX
soc/soc_legacy/arm/nxp_imx/mcimx7_m4/Kconfig.series:    select SOC_FAMILY_IMX
soc/soc_legacy/arm/nxp_imx/mimx8ml8_m7/Kconfig.series:  select SOC_FAMILY_IMX
soc/soc_legacy/arm/nxp_imx/mimx8mm6_m4/Kconfig.series:  select SOC_FAMILY_IMX
soc/soc_legacy/arm/nxp_imx/mimx8mq6_m4/Kconfig.series:  select SOC_FAMILY_IMX
soc/soc_legacy/arm/nxp_imx/rt/Kconfig.series:   select SOC_FAMILY_IMX
soc/soc_legacy/arm/nxp_imx/rt5xx/Kconfig.series:        select SOC_FAMILY_IMX
soc/soc_legacy/arm/nxp_imx/rt6xx/Kconfig.series:        select SOC_FAMILY_IMX